content/03-reference/01-tools-and-interfaces/02-prisma-client/090-raw-database-access.mdx

```diff
@@ -26,71 +44,205 @@ const result = await prisma.queryRaw('SELECT * FROM User;')
+
+### Return type
 
-## Tagged templates
+`executeRaw` returns a `number`.
 
-Both the `queryRaw` and `executeRaw` methods are implemented as a [tagged template](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Tagged_templates). Therefore, you can also call `queryRaw` as follows:
+### Signature
 
 ```ts
-const result = await prisma.queryRaw`SELECT * FROM User;`
+executeRaw<T = any>(query: string | TemplateStringsArray, ...values: any[]): Promise<number>;
 ```
 
-`executeRaw` can also be called like so:
+## Using variables
+
+`executeRaw` and `queryRaw` are implemented as [**tagged templates**](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Tagged_templates). Tagged templates are the recommended way to use variables with raw SQL in Prisma Client.
+
+The following example includes a placeholder named `${userId}`. Note the double backticks (``` `` ```) instead of `()`:
 
 ```ts
-const result: number = await prisma.executeRaw`UPDATE User SET active = ${true} WHERE emailValidated = ${true}`
+const userId = 42
+const result = await prisma.queryRaw`SELECT * FROM User WHERE id = ${userId};` 
 ```
 
-## Setting variables
+> **Important**: You *must* use double backticks - the same query using `queryRaw()` is not secure because `${userId}` is not escaped.
+
+✔ Benefits of using the tagged template versions of `queryRaw` and `executeRaw` include:
+
+* Prisma Client escapes all variables.
+* Tagged templates are database-agnostic - you do not need to remember if variables should be written as `$1` (PostgreSQL) or `?` (MySQL).
+* [SQL Template Tag](https://github.com/blakeembrey/sql-template-tag) give you access to [useful helpers](#tagged-template-helpers).
+* Embedded, named variables are easier to read.
+
+> **Note**: You cannot pass a table or column name into a tagged template placeholder. For example, you cannot `SELECT ?` and pass in `*` or `id, name` based on some condition.
+
+### Tagged template helpers
 
-To include variables in your SQL query, you can use JavaScript string interpolation with [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals):
+Prisma Client specifically uses [SQL Template Tag](https://github.com/blakeembrey/sql-template-tag
+), which exposes a number of helpers. For example, the following query uses `join()` to pass in a list of IDs:
 
 ```ts
-const userId = 42
-const result = await prisma.queryRaw`SELECT * FROM User WHERE id = ${userId};`
+import { join } from "@prisma/client";
+
+const ids = [1, 3, 5, 10, 20];
+const result = await prisma.queryRaw`SELECT * FROM User WHERE id IN (${join(
+  ids
+)})`;
+```
+
+The following example uses the `empty` and `sql` helpers to change the query depending on whether `userName` is empty:
+
+```ts
+import { sql, empty } from "@prisma/client";
+
+const userName = "";
+const result = await prisma.queryRaw`SELECT * FROM User ${
```

Comment:
Could be cool to comment the "generated" SQL maybe? That may look a bit cryptic to read only through code.